### PR TITLE
fix for 843

### PR DESCRIPTION
**Pull request by rhsakarpos**
**What this PR does / why we need it**:
this fixes issue 843

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
#843 

**Special notes for your reviewer**:
image

image

**Release note**:

```
Remove the mysql container. Install the mysql client in s3 container. Add a script in the s3 container to connect to tidb and initialize it (with a retry mechanism, to retry 5 times, 2 seconds apart), then start the s3 process
```

### CODE REVIEW

**Comment by rhsakarpos** [comment]
@kumarashit @sfzeng @nguptaopensds please review, thanks

File: s3/initdb.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+succeed=false
+for i in 1 2 3 4 5
+do
+    echo  "trying again, attempt " $i
+    mysql -h tidb -P 4000 -u root -e "create database if not exists s3 character set utf8;use s3;source ./s3.sql;"
+    if [ "$?" = "0" ]; then
+        succeed=true
+        break
+    fi
+    sleep 5
```
Comment (sfzeng) [inline s3/initdb.sh]: Is it better to seep less time for each loop round? Say 1 second.

File: s3/initdb.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    echo  "sleeping to try again"
```
Comment (sfzeng) [inline s3/initdb.sh]: Move this statement before sleep, please.

File: s3/initdb.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+#host=$(ip route show | grep "default" | awk '{print $3}' | xargs echo -n)
+#echo $host
+succeed=false
+for i in 1 2 3 4 5
+do
+    echo  "trying again, attempt " $i
+    mysql -h tidb -P 4000 -u root -e "create database if not exists s3 character set utf8;use s3;source ./s3.sql;"
```
Comment (sfzeng) [inline s3/initdb.sh]: It's supposed to be: mysql -h tidb  -P 4000 -u root -e "create database if not exists s3 character set utf8;use s3;source /s3.sql;create database if not exists yig character set utf8;use yig;source /yig.sql;"

File: docker-compose.yml [hunk 4]
```diff
@@ -133,14 +135,4 @@ services:
...
-    container_name: mysql
-    volumes:
-      - ./s3/pkg/conf/tidb.sql:/s3.sql
-      - ./s3/pkg/datastore/yig/conf/yig.sql:/yig.sql
-    command: mysql -h tidb  -P 4000 -u root -e "create database if not exists s3 character set utf8;use s3;source /s3.sql;create database if not exists yig character set utf8;use yig;source /yig.sql;"
-    environment:
-      - MYSQL_ALLOW_EMPTY_PASSWORD=yes
-    depends_on:
-      - tidb
+
```
Comment (sfzeng) [inline docker-compose.yml]: Please remove those useless blank line, thanks.

File: s3/initdb.sh (hunk 1 above)
Comment (rhsakarpos) [reply s3/initdb.sh]: okay, 1s maybe too low, lead to multiple tries, keep it at 2 seconds (based on testing, with 1 second, it tried 2 or 3 times before success)

File: s3/initdb.sh (hunk 2 above)
Comment (rhsakarpos) [reply s3/initdb.sh]: done

File: s3/initdb.sh (hunk 3 above)
Comment (rhsakarpos) [reply s3/initdb.sh]: added

File: s3/initdb.sh (hunk 1 above)
Comment (sfzeng) [reply s3/initdb.sh]: OK, thanks.

File: s3/Dockerfile [hunk 5]
```diff
@@ -19,9 +19,12 @@ RUN rpm --import https://packages.confluent.io/rpm/5.2/archive.key && yum makeca
 yum install -y epel-release && \
 rpm -ivh https://download.ceph.com/rpm-luminous/el7/noarch/ceph-release-1-1.el7.noarch.rpm && \
 yum --enablerepo=epel-testing install -y lttng-ust libradosstriper-devel librados2-devel librdkafka-devel && \
+yum -y install mysql && \
 yum clean all && \
 rm -rf /var/cache/yum
```
Comment (sfzeng) [inline s3/Dockerfile]: Could you please remove those extra blank lines?

Comment (rhsakarpos) [reply s3/Dockerfile]: done

**Pull request merged**